doc/man3/flux_watcher_set_priority.rst

```diff
@@ -0,0 +1,42 @@
+============================
+flux_watcher_set_priority(3)
+============================
+
+.. default-domain:: c
+
+SYNOPSIS
+========
+
+.. code-block:: c
+
+   #include <flux/core.h>
+
+   flux_watcher_t *flux_watcher_set_priority (flux_watcher_t *w,
+                                              int priority);
+
+Link with :command:`-lflux-core`.
+
+DESCRIPTION
+===========
+
+:func:`flux_watcher_set_priority` sets the priority on the watcher.
+Higher priority watchers run first.  The range of priorities is from
+-2 to 2, with the default being 0.  If the priority is out
+of range, the max or min value is set.
+
+This function is a no-op if the underlying watcher does not support priorities.
+Currently only the check watcher supports priorities.
```

Comment:
Oh, one thing: this should mention that the priority can only be set when the watcher is stopped.

Reply:
oh good catch and should be in the function comments too.

```
diff --git a/doc/man3/flux_watcher_set_priority.rst b/doc/man3/flux_watcher_set_priority.rst
index 8138330cb..dfc507e10 100644
--- a/doc/man3/flux_watcher_set_priority.rst
+++ b/doc/man3/flux_watcher_set_priority.rst
@@ -22,7 +22,8 @@ DESCRIPTION
 :func:`flux_watcher_set_priority` sets the priority on the watcher.
 Higher priority watchers run first.  The range of priorities is from
 -2 to 2, with the default being 0.  If the priority is out
-of range, the max or min value is set.
+of range, the max or min value is set.  The priority should only be
+set when the watcher is stopped.
 
 This function is a no-op if the underlying watcher does not support priorities.
 Currently only the check watcher supports priorities.
diff --git a/src/common/libflux/reactor.h b/src/common/libflux/reactor.h
index 69e9ebfef..f29d8c57f 100644
--- a/src/common/libflux/reactor.h
+++ b/src/common/libflux/reactor.h
@@ -76,6 +76,7 @@ typedef void (*flux_watcher_f)(flux_reactor_t *r, flux_watcher_t *w,
  * Higher priority watchers run first.
  * This is a no-op if the underlying watcher doesn't support it.
  * If the priority is out of range, the max or min value is set.
+ * The priority should only be set when the watcher is stopped.
  * Currently only the check watcher supports it.
  */
 void flux_watcher_set_priority (flux_watcher_t *w, int priority);

```